theatre/core/src/sheetObjects/TheatreSheetObject.ts

```diff
@@ -153,8 +160,27 @@ export default class TheatreSheetObject<
     return this._valuesDerivation().tapImmediate(coreTicker, fn)
   }
 
+  // internal: Make the deviration keepHot if directly read
   get value(): PropsValue<Props> {
-    return this._valuesDerivation().getValue()
+    const der = this._valuesDerivation()
+    if (KEEP_HOT_FOR_MS != null) {
+      if (!der.isHot && this._keepHotSchedule == null) {
+        const untap = der.keepHot()
+        const clear = () => {
+          if (this._keepHotSchedule?.clear === clear) {
+            this._keepHotSchedule = undefined
+            untap()
+          }
+        }
+        this._keepHotSchedule = {
+          schedule: debounce(clear, KEEP_HOT_FOR_MS),
+          clear,
+        }
+      }
+
+      this._keepHotSchedule?.schedule()
+    }
+    return der.getValue()
```

Comment:
I rewrote this function to understand it, came up with this equivalent(?) code which has a few less moving parts:

```suggestion
private _keepHotDebouncedUntap: undefined | VoidFn
// ...
 const der = this._valuesDerivation()
    if (KEEP_HOT_FOR_MS != null) {
      if (!der.isHot && this._keepHotDebouncedUntap == null) {
        const untap = der.keepHot()
        this._keepHotDebouncedUntap = debounce(untap, KEEP_HOT_FOR_MS)
      }

      this._keepHotDebouncedUntap?.()
    }
    return der.getValue()
```
```

Reply:
I'll have to look at this closer. Thanks for the suggestion. I definitely didn't love the feeling of how this was written before. It does seem simpler to just use a debounce helper

Reply:
LMK what you think of the new approach